modepy/modes.py

```diff
@@ -658,6 +658,10 @@ def tensor_product_basis(dims, basis_1d):
 
     .. versionadded:: 2017.1
     """
+    if dims == 0:
+        # NOTE: using to maintain consistency in the 0d case
+        return simplex_onb(dims, len(basis_1d))
```

Comment:
This yuck has to go. Part of why I'm eager to have a new API (the `shapes` thing provides a convenient excuse) is to get rid of this. (Another motivation for a new API is to use symbolic basis functions.)

(Feel free to resolve this without action, but I had to say it.)